pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java

```diff
@@ -83,9 +83,9 @@ public void onManagedLedgerPropertiesInitialize(Map<String, String> propertiesMa
     public void onManagedLedgerLastLedgerInitialize(String name, LedgerHandle lh) {
         try {
             for (BrokerEntryMetadataInterceptor interceptor : brokerEntryMetadataInterceptors) {
-                if (interceptor instanceof AppendIndexMetadataInterceptor) {
+                if (interceptor instanceof AppendIndexMetadataInterceptor && lh.getLastAddConfirmed() >= 0) {
```

Comment:
Using instanceof is a bad practice, not Object oriented.

It is not in the scope of this PR but it would be better to have a more OO approach, using polymorphism

Reply:
Agree with you that instanceof is not a good practice if we have a better choice. 
There are two class `AppendIndexMetadataInterceptor` and `AppendBrokerTimestampMetadataInterceptor` implement `BrokerEntryMetadataInterceptor`. 
If using polymorphism and the interceptor is a `AppendBrokerTimestampMetadataInterceptor` whose `recoveryIndexGenerator` method do nothing like 
```java
public void recoveryIndexGenerator(long index) {
       // do nothing.
    }
```
, we still meed to deserialize the `BrokerEntryMetadata`  from entry.

Reply:
/pulsarbot run-failure-checks